src/credential.rs

```diff
@@ -74,6 +74,22 @@ impl Default for CredentialFlat {
 }
 
 impl CredentialFlat {
+    pub fn get_properties_byte(&self) -> u8 {
+        let mut res: u8 = 0;
+        res |= if self.touch_required { 1 << 0 } else { 0 };
+        res |= if self.encryption_key_type.unwrap() == EncryptionKeyType::PinBased {
+            1 << 1
+        } else {
+            0
+        };
+        res |= if self.login.is_some() || self.password.is_some() {
+            1 << 2
+        } else {
+            0
+        };
```

Comment:
Rather than bit manipulation I'd use the bitflags crate but this works.

Reply:
I was wondering about it, but did not want to introduce another dependency (I've missed it is used). Will keep in mind to migrate to it on the next occasion.